src/main/drivers/accgyro/accgyro_spi_icm42688p.c

```diff
@@ -0,0 +1,283 @@
+#define ICM42688P_INT1_DRIVE_CIRCUIT_OD              (0 << 1)
+#define ICM42688P_INT1_DRIVE_CIRCUIT_PP              (1 << 1)
+#define ICM42688P_INT1_POLARITY_ACTIVE_LOW           (0 << 0)
+#define ICM42688P_INT1_POLARITY_ACTIVE_HIGH          (1 << 0)
+
+#define ICM42688P_RA_INT_CONFIG0                     0x63
+#define ICM42688P_UI_DRDY_INT_CLEAR_ON_SBR           ((0 << 5) || (0 << 4))
+#define ICM42688P_UI_DRDY_INT_CLEAR_ON_SBR_DUPLICATE ((0 << 5) || (1 << 4)) // duplicate settings in datasheet, Rev 1.3.
+#define ICM42688P_UI_DRDY_INT_CLEAR_ON_F1BR          ((1 << 5) || (0 << 4))
+#define ICM42688P_UI_DRDY_INT_CLEAR_ON_SBR_AND_F1BR  ((1 << 5) || (1 << 4))
+
+#define ICM42688P_RA_INT_CONFIG1                     0x64
+#define ICM42688P_INT_ASYNC_RESET_BIT                4
+#define ICM42688P_INT_TDEASSERT_DISABLE_BIT          5
+#define ICM42688P_INT_TDEASSERT_ENABLED              (0 << ICM42688P_INT_TDEASSERT_DISABLE_BIT)
+#define ICM42688P_INT_TDEASSERT_DISABLED             (1 << ICM42688P_INT_TDEASSERT_DISABLE_BIT)
+#define ICM42688P_INT_TPULSE_DURATION_BIT            6
+#define ICM42688P_INT_TPULSE_DURATION_100            (0 << ICM42688P_INT_TPULSE_DURATION_BIT)
+#define ICM42688P_INT_TPULSE_DURATION_8              (1 << ICM42688P_INT_TPULSE_DURATION_BIT)
+
+
+#define ICM42688P_RA_INT_SOURCE0                     0x65
+#define ICM42688P_UI_DRDY_INT1_EN_DISABLED           (0 << 3)
+#define ICM42688P_UI_DRDY_INT1_EN_ENABLED            (1 << 3)
+
+static void icm42688PSpiInit(const busDevice_t *bus)
+{
+    static bool hardwareInitialised = false;
+
+    if (hardwareInitialised) {
+        return;
+    }
+
+
+    spiSetDivisor(bus->busdev_u.spi.instance, spiCalculateDivider(ICM42688P_MAX_SPI_CLK_HZ));
+
+    hardwareInitialised = true;
+}
+
+uint8_t icm42688PSpiDetect(const busDevice_t *bus)
+{
+    icm42688PSpiInit(bus);
+
+    spiSetDivisor(bus->busdev_u.spi.instance, spiCalculateDivider(ICM42688P_MAX_SPI_INIT_CLK_HZ));
+
+    spiBusWriteRegister(bus, ICM42688P_RA_PWR_MGMT0, 0x00);
+
+    uint8_t icmDetected = MPU_NONE;
+    uint8_t attemptsRemaining = 20;
+    do {
+        delay(150);
+        const uint8_t whoAmI = spiBusReadRegister(bus, MPU_RA_WHO_AM_I);
+        switch (whoAmI) {
+        case ICM42688P_WHO_AM_I_CONST:
+            icmDetected = ICM_42688P_SPI;
+            break;
+        default:
+            icmDetected = MPU_NONE;
+            break;
+        }
+        if (icmDetected != MPU_NONE) {
+            break;
+        }
+        if (!attemptsRemaining) {
+            return MPU_NONE;
+        }
+    } while (attemptsRemaining--);
+
+    spiSetDivisor(bus->busdev_u.spi.instance, spiCalculateDivider(ICM42688P_MAX_SPI_CLK_HZ));
+
+    return icmDetected;
+}
+
+void icm42688PAccInit(accDev_t *acc)
+{
+    acc->acc_1G = 512 * 4;
+}
+
+bool icm42688PAccRead(accDev_t *acc)
+{
+    uint8_t data[6];
+
+    const bool ack = busReadRegisterBuffer(&acc->bus, ICM42688P_RA_ACCEL_DATA_X1, data, 6);
```

Comment:
```suggestion
    if (!busReadRegisterBuffer(&acc->bus, ICM42688P_RA_ACCEL_DATA_X1, data, 6)) {
        return false;
    }
```

Reply:
You'll find that when I write boolean expressions I name the booleans to make the code easier to understand and read aloud.  i.e. "if not ack" ...

I use a similar pattern in many other places for code clarity.  The optimizer will optimize out the local variables and when debugging more complicated boolean expressions it really helps.  When operator precedence and short-circuiting is required then I will use a well-named method call.

This is also not new code, it's in the ICM42605 driver too.

Reply:
@hydra yes the compiler should make no difference here. Readability is definite a good thing. Thanks for the reminder.